core/transform.h

```diff
@@ -13,6 +13,56 @@ namespace scipp::core {
 
 namespace detail {
 
+template <class T> struct ValueAndVariance {
+  T value;
+  T variance;
+};
+
+template <class T1, class T2>
+constexpr auto operator+(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value + b.value, a.variance + b.variance};
+}
+template <class T1, class T2>
+constexpr auto operator-(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value - b.value, a.variance - b.variance};
+}
+template <class T1, class T2>
+constexpr auto operator*(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value * b.value,
+                          a.variance * b.value * b.value +
+                              b.variance * a.value * a.value};
+}
+template <class T1, class T2>
+constexpr auto operator/(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value / b.value,
+                          a.variance / (b.value * b.value) +
+                              b.variance / (a.value * a.value)};
```

Comment:
Didn't we find a better way of writing this to avoid NaNs in case either `a` or `b` is zero?

Reply:
Thanks for checking this. It turns out I had actually mixed up relative and absolute errors, so the new unit test was also wrong -_-.